src/braintree-manager/payment-providers/apple-pay/apple-pay-session-datasource.ts

```diff
@@ -0,0 +1,128 @@
+import { DonationPaymentInfo } from '../../../models/donation-info/donation-payment-info';
+import { BraintreeManagerInterface } from '../../braintree-interfaces';
+import { BillingInfo } from '../../../models/common/billing-info';
+import { CustomerInfo } from '../../../models/common/customer-info';
+import { DonationRequestCustomFields } from '../../../models/request-models/donation-request';
+import { DonationResponse } from '../../../models/response-models/donation-response';
+import { PaymentProvider } from '../../../models/common/payment-provider-name';
+
+export interface ApplePaySessionDataSourceInterface {
+  delegate?: ApplePaySessionDataSourceDelegate;
+  donationInfo: DonationPaymentInfo;
+  onvalidatemerchant(event: ApplePayJS.ApplePayValidateMerchantEvent): Promise<void>;
+  onpaymentauthorized(event: ApplePayJS.ApplePayPaymentAuthorizedEvent): Promise<void>;
+  oncancel(): Promise<void>;
+}
+
+export interface ApplePaySessionDataSourceDelegate {
+  paymentComplete(response: DonationResponse): void;
+  paymentFailed(error: string): void;
+  paymentCancelled(): void;
+}
+
+export class ApplePaySessionDataSource implements ApplePaySessionDataSourceInterface {
+  delegate?: ApplePaySessionDataSourceDelegate;
+  donationInfo: DonationPaymentInfo;
+
+  private session: ApplePaySession;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  private applePayInstance: any;
+  private braintreeManager: BraintreeManagerInterface;
+
+  constructor(options: {
+    donationInfo: DonationPaymentInfo;
+    session: ApplePaySession;
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    applePayInstance: any;
```

Comment:
something weird in its return?

Reply:
Are you talking about the `any`? 

We don't have TypeScript declarations for some of the third part libraries (Apple Pay in this case) so we have to assign them `any` so TypeScript ignores them. We have an eslint rule to not do this since this isn't really good practice so I you can disable that when needed.

---

This is the problem that [`DefinitelyTyped`](https://github.com/DefinitelyTyped/DefinitelyTyped) is trying to solve.